docs/content/040-adoption-guides/040-nexus-with-nextjs.md

```diff
@@ -0,0 +1,328 @@
+
+```ts
+// pages/api/graphql.ts
+
+import { makeSchema, queryType } from '@nexus/schema'
+import { ApolloServer } from 'apollo-server-micro'
+
+const Query = queryType({
+  definition(t) {
+    t.string('hello', { resolve: () => 'hello world!' })
+  },
+})
+
+const schema = makeSchema({
+  types: [Query],
+})
+
+const server = new ApolloServer({
+  schema,
+})
+
+export const config = {
+  api: {
+    bodyParser: false,
+  },
+}
+
+export default server.createHandler({
+  path: '/api/graphql',
+})
+```
+
+Now if we navigate to `localhost:3000/api/graphql`, we should see the GraphQL Playground and we should be able to make a query for the 'hello world!' message.
+
+![graphql playground with the hello world message](/assets/next-nexus-1.png)
+
+## Create a Schema
+
+When creating a schema with Nexus, we have the option of housing all the parts of the schema in a single file or we can split it out into many files. In either case, it's tempting to place these files near the `graphql.ts` API route. However, since Next.js will create an API route for every file in the `pages/api` directory, we should take care **not** to do this.
+
+It's best to keep all the parts of your schema in a separate directory. Most often, this will be a directory at the root of your project.
+
+Create a diretory called `schema` at the project root.
+
+```bash
+mkdir schema
+```
+
+Let's start by creating a schema to retrieve some data with the `queryType`.
+
+in the `schema` directory, create a file called `Query.ts`.
+
+```bash
+cd schema
+touch Query.ts
+```
+
+At the very least, this file should export a `queryType` that can be used in our schema.
+
+Let's create and export a `Query` constant, as well as a `Framework` object type to represent a set of front end technologies.
+
+```ts
+// schema/Query.ts
+
+import { objectType, queryType } from '@nexus/schema'
+
+export const Framework = objectType({
+  name: 'Framework',
+  definition(t) {
+    t.id('id')
+    t.string('name')
+  },
+})
+
+export const Query = queryType({
+  definition(t) {
+    t.list.field('frameworks', {
+      type: 'Framework',
+      resolve: () => {
+        return [
+          {
+            id: '1',
+            name: 'React',
+          },
+          {
+            id: '2',
+            name: 'Vue',
+          },
+          {
+            id: '3',
+            name: 'Angular',
+          },
+          {
+            id: '4',
+            name: 'Svelte',
+          },
+        ]
+      },
+    })
+  },
+})
+```
+
+Let's now create a barrel file in the `schema` directory which will be responsible for importing all the types in the directory and exporting a constructed schema using `makeSchema`.
+
+```bash
+cd schema
+touch index.ts
+```
+
+In the `index.ts` file, bring in `makeSchema` as well as the types we just created in `Query.ts`.
+
+```ts
+// schema/index.ts
+
+import { makeSchema } from '@nexus/schema'
+import { Framework, Query } from './Query'
```

Comment:
```suggestion
import * as QueryTypes from './Query'
```

Would it be simpler like this so that there is a direct passthrough established between the module exports and the makeSchema config?

Reply:
I favor this change too. This is considered a best practice of nexus and should be used as much as possible in our examples/tutorial to teach this pattern